examples/oauth-authenticate.hs

```diff
@@ -0,0 +1,67 @@
+#! /usr/bin/env stack
+{- stack --resolver lts-10.4 --install-ghc runghc --package classy-prelude --package http-client --package crypto-random --package oauthenticated --package uri-templater -}
+{-# OPTIONS_GHC -Wall -Werror           #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NoImplicitPrelude          #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE PackageImports             #-}
+{-# LANGUAGE QuasiQuotes                #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+
+import ClassyPrelude
+import qualified "crypto-random" Crypto.Random as Crypto
+import Data.Aeson (FromJSON, Value, eitherDecodeStrict')
+import qualified Data.ByteString.Char8 as Char8
+import qualified Network.HTTP.Client as Client
+import qualified Network.OAuth as OAuth
+import qualified Options.Applicative as Opt
+
+data Opts = Opts
+  { oauthUrl    :: String
+  , oauthKey    :: ByteString
+  , oauthSecret :: ByteString
+  }
+
+data App = App
+  { appRng     :: MVar Crypto.SystemRNG
+  , appManager :: Client.Manager
+  , appServer  :: OAuth.Server
+  , appCred    :: OAuth.Cred OAuth.Client
+  }
+
+signRequest :: App -> Client.Request -> IO Client.Request
```

Comment:
Can this be something like `StateT SystemRNG IO Client.Request` and take the rng out of App? Not a big deal, but bringing MVar into it seems unnecessary for an example.

Reply:
👍